src/utils/api/transaction/lsk.js

```diff
@@ -1,11 +1,144 @@
+const txFilters = {
+  dateFrom: { key: 'from', test: str => typeof str === 'string' },
+  dateTo: { key: 'to', test: str => typeof str === 'string' },
+  amountFrom: { key: 'min', test: str => typeof str === 'string' },
+  amountTo: { key: 'max', test: str => typeof str === 'string' },
+  limit: { key: 'limit', test: num => (typeof num === 'number') },
+  offset: { key: 'offset', test: num => (typeof num === 'number' && num > 0) },
+  sort: {
+    key: 'sort',
+    test: str => ['amount:asc', 'amount:desc', 'timestamp:asc', 'timestamp:desc'].includes(str),
+  },
+};
+/**
+ * Retrieves the list of transactions for given parameters
+ *
+ * @param {Object} data
+ * @param {Object} data.network - Network setting from Redux store
+ * @param {String?} data.baseUrl - Lisk Service API url to override the
+ * existing ServiceUrl on the network param. We may use this to retrieve
+ * the details of an archived transaction.
+ * @param {Object} data.params
+ * @param {String} data.params.dateFrom Unix timestamp, the start time of txs
+ * @param {String} data.params.dateTo Unix timestamp, the end time of txs
+ * @param {String} data.params.amountFrom The minimum value of txs
+ * @param {String} data.params.amountTo The maximum value of txs
+ * @param {String} data.params.type The title of the transaction type
+ * @param {Number} data.params.offset Used for pagination
+ * @param {Number} data.params.limit Used for pagination
+ * @param {String} data.params.sort an option of 'amount:asc',
+ * 'amount:desc', 'timestamp:asc', 'timestamp:desc',
+ * @param {Object} data.params.blockId The id of the block whose transaction we want.
+ * If passed, all other parameter will be ignored.
+ * @returns {Promise} Transactions list API call
+ */
+export const getTransactions = ({
+  network,
+  params,
+  baseUrl,
+}) => {
+  const typeConfig = transactionTypes()[params.type];
+  // if type, correct the type and use WS
+  if (params.type && typeConfig) {
+    const requests = Object.values(typeConfig.code).map(type => ({
+      method: 'get.transactions',
+      params: { type },
+    }));
+    // BaseUrl is only used for retrieving archived txs, so it's not needed here.
+    return ws({ baseUrl: network.serviceUrl, requests });
+  }
 
-export const getTransactionStats = data => new Promise(resolve =>
-  resolve({ endpoint: 'getTransactionStats', token: 'LSK', data }));
+  const normParams = {};
+
+  // if blockId, ignore others
+  if (params.blockId) {
+    normParams.block = params.blockId;
+  } else {
+    // Validate params and fix keys
+    Object.keys(params).forEach((key) => {
+      if (txFilters[key].test(params[key])) {
+        normParams[txFilters[key].key] = params[key];
+      }
+    });
+  }
+
+  return http({
+    network,
+    path: 'transactions',
+    params: normParams,
+    baseUrl,
+  });
+};
+
+export const getRegisteredDelegates = async ({ network }) => {
+  const delegates = await getDelegates({
+    network,
+    params: { limit: 1 },
```

Comment:
Why limit 1? Won't this resolve in a list of 1 delegate?

Reply:
I only need ‘response.meta.total’. I don’t use the retrieved delegate info